### DISCUSSION TITLE
How to custom form request response

### DISCUSSION THREAD

**Discussion by rickyanwar** [MarcinOrlowski/laravel-api-response-builder#241] [Q&A]
How to custom form request response 
laravel form request has default HTTP code 422, can we adjust with this library?

**Comment by MarcinOrlowski** [comment]
Can you please post relevant snippet of the code that you currently use, so we can have better context?

**Reply by rickyanwar** [reply]
i don't find way to adjust with this library,so i just override in Handler Exepction

```

  if ($request->is('api/*')) {
                return response()->json(
                    [
                        "success" => false,
                        "code" => 422,
                        "message" =>"PRECONDITION_FAILED",
                        "data" => ['errors' => $e->errors()]
                    ], 422
                );
            }
```

**Reply by MarcinOrlowski** [reply]
That's proper approach because of how Laravel deals with validation errors. But this is also covered. Please see [Exception handling](https://github.com/MarcinOrlowski/laravel-api-response-builder/blob/master/docs/exceptions.md) docs for more details.

**Reply by rickyviz** [reply]
how about a custom request success response ?
 i try to override my controller with this code, but api code still return 0, and i can't change message

   ```
 public function respond($data = null, $message = [], $api_code = 200)
    {
        return ResponseBuilder::success($data, $api_code, $message );
    }
```

**Reply by MarcinOrlowski** [reply]
No plan for this. Success is always `0` by design. I might consider adding `warning` state but I am skeptical this is proper approach as this makes simple thing complicated. Currently you can have two states - either succes (`0`) or error (non `0`). Do you **really** need more? I would say that most likely you do not. Either action succeeded or failed. If succeeded  with notes, then return notes in the payload but that still `success` type of response. Do I miss anything here?

**Reply by rickyviz** [reply]
i need a success function like error WithMessageAnd Data with three parameters 
`($api code, $message, $data, $http_code = null)`, but throw success

any idea?

**Reply by MarcinOrlowski** [reply]
Sorry for late answer but you already can do that by existing API
